kitsune/community/api.py

```diff
@@ -0,0 +1,330 @@
+from collections import defaultdict
+from datetime import datetime, timedelta
+
+from elasticutils import F
+from rest_framework import views, fields
+from rest_framework.response import Response
+
+from kitsune.questions.models import AnswerMetricsMappingType
+from kitsune.sumo.api import GenericAPIException
+from kitsune.users.models import UserMappingType
+from kitsune.wiki.models import RevisionMetricsMappingType
+
+
+# This should be the higher than the max number of contributors for a
+# section.  There isn't a way to tell ES to just return everything.
+BIG_NUMBER = 10000
+
+
+class TopContributorsBase(views.APIView):
+
+    def get(self, request):
+        return Response(self.get_data(request))
+
+    def get_filters(self):
+        self.filter_values = self.get_default_filters()
+        # request.GET is a multidict, so simple `.update(request.GET)` causes
+        # everything to be a list. This converts it into a plain single dict.
+        self.filter_values.update(dict(self.request.GET.items()))
+
+        f = F()
+
+        for key, value in self.filter_values.items():
+            method_name = 'filter_' + key
+            if not hasattr(self, method_name):
+                raise GenericAPIException(400, 'Unknown filter field {}'.format(key))
+            filter_method = getattr(self, 'filter_' + key)
```

Comment:
This doesn't matter a whole  lot, but it's very slightly better to do this:
```
filter_method = getattr(self, 'filter_' + key, None)
if filter_method is None:
    ...
```

The Python docs say that hasattr just turns around and calls getattr, catches the exception and returns True/False accordingly:

https://docs.python.org/2/library/functions.html#hasattr

So given that you want to do something with the value, it's slightly better to just use getattr.

Having said that, it doesn't meaningfully matter one way or the other. Mostly I mention it in case it's interesting.